activerecord/CHANGELOG.md

```diff
@@ -1,3 +1,33 @@
+*   Run transactional callbacks on the freshest instance to save a given 
+    record within a transaction.
+
+    When multiple Active Record instances change the same record within a 
+    transaction, Rails runs `after_commit` or `after_rollback` callbacks for 
+    only one of them. This adds an option to specify how Rails chooses which 
+    instance receives the callbacks. It also changes the framework defaults to 
+    use the new logic.
+
+    When `config.active_record.run_commit_callbacks_on_first_saved_instances_in_transaction` 
+    is `true`, transactional callbacks are run on the first instance to save, 
+    even though its instance state may be stale.
+
+    When it is `false`, which is the new framework default starting with version 
+    7.1, transactional callbacks are run on the instances with the freshest 
+    instance state. Those instances are chosen as follows:
+
+    - In general, run transactional callbacks on the last instance to save a 
+      given record within the transaction.
+    - There are two exceptions:
+        - If the record is created within the transaction, then 
+          `after_create_commit` callbacks will be fired on the last instance to 
+          save, even if that instance performed an update.
```

Comment:
Is this an exception? It is running the callback on the last instance to save.

Reply:
The exceptional behavior is that we're running `after_create_commit`, not `after_update_commit` on that instance, even though from its perspective it just issued an update

https://github.com/rails/rails/blob/ad54b7a6733373b5426ceed6c4663f99e9781dbb/activerecord/test/cases/transaction_callbacks_test.rb#L791-L802

If you don't feel like that detail needs to be called out, then I'm happy to remove or reword.

Reply:
Ahhh, got it, that is not clear in the message. Can we update the message to show that difference?

Reply:
Okay, I updated it. Is that more clear?